surefire-providers/surefire-testng/src/main/java/org/apache/maven/surefire/testng/conf/TestNG740Configurator.java

```diff
@@ -0,0 +1,72 @@
+package org.apache.maven.surefire.testng.conf;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.maven.surefire.api.testset.TestSetFailedException;
+import org.testng.xml.XmlSuite;
+
+import java.lang.reflect.Method;
+import java.util.Map;
+
+import static java.lang.Integer.parseInt;
+import static org.apache.maven.surefire.api.booter.ProviderParameterNames.PARALLEL_PROP;
+import static org.apache.maven.surefire.api.booter.ProviderParameterNames.THREADCOUNT_PROP;
+
+/**
+ * TestNG 7.4.0 configurator. Changed setParallel type to enum value.
+ * Uses reflection since ParallelMode enum doesn't exist in supported
+ * TestNG 5.x versions.
+ *
+ * @since 3.0.0-M6
+ */
+public class TestNG740Configurator extends TestNG60Configurator
+{
+    @Override
+    public void configure( XmlSuite suite, Map<String, String> options )
+        throws TestSetFailedException
+    {
+        String threadCountAsString = options.get( THREADCOUNT_PROP );
+        int threadCount = threadCountAsString == null ? 1 : parseInt( threadCountAsString );
+        suite.setThreadCount( threadCount );
+
+        String parallel = options.get( PARALLEL_PROP );
+        if ( parallel != null )
+        {
+            try
+            {
+                Class enumClass = Class.forName( "org.testng.xml.XmlSuite$ParallelMode" );
+                Method resolverMethod = enumClass.getMethod( "getValidParallel", String.class );
```

Comment:
Pls do not use reflection. We use `org.apache.maven.surefire.api.util.ReflectionUtils` in Surefire.
But in this case I would use the static method `valueOf()` on `java.lang.Enum` and upper case the string this way in a single line:
```
Object parallelEnum = ( ( Enum ) enumClass ).valueOf(  parallel.toUpperCase() );
```
https://github.com/cbeust/testng/blob/master/src/main/java/org/testng/xml/XmlSuite.java#L19
If `parallel.toUpperCase()` is other than null, "CLASSES" and "METHODS", then throw `TestSetFailedException`. Can you please extend the Javadoc with one more sentence saying that TestNG supports parallel with two values "classes" and "methods", see https://github.com/apache/maven-surefire/blob/master/maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java#L577.
If you want to add "instances", then we have to open another PR because there are some validation checks   and conditions which need to be updated.

Reply:
ok, I can update this to use `Enum.valueOf()`, restrict the allowable values and update the javadoc.
Is the `Class.forName` ok to get `enumClass`?  If not, how should this work given the class doesn't exist in TestNG < 6.9.7?
I can also use `ReflectionUtils` to call the appropriate `setParallel` method instead of raw reflection, is that ok or is there a better way?

Reply:
> 
> 
> ok, I can update this to use `Enum.valueOf()`, restrict the allowable values and update the javadoc.
> Is the `Class.forName` ok to get `enumClass`? If not, how should this work given the class doesn't exist in TestNG < 6.9.7?

I think this will work as you have done it. Example, this is similar we already use in another class in the same TestNG provider
```
Class<?> reporterConfig = Class.forName( "org.testng.ReporterConfig" )
```

> I can also use `ReflectionUtils` to call the appropriate `setParallel` method instead of raw reflection, is that ok or is there a better way?

I have already done it for you, see the next comment.